### PR TITLE
Rustup to https://github.com/rust-lang/rust/pull/65426

### PR DESCRIPTION
**Pull request by phansch**
changelog: none

### CODE REVIEW

File: clippy_lints/src/booleans.rs [hunk 1]
```diff
@@ -255,7 +256,7 @@ fn simplify_not(cx: &LateContext<'_, '_>, expr: &Expr) -> Option<String> {
                 .iter()
                 .cloned()
                 .flat_map(|(a, b)| vec![(a, b), (b, a)])
-                .find(|&(a, _)| a == path.ident.name.as_str())
+                .find(|&(a, _)| Symbol::intern(a) == path.ident.name)
```
Comment (lzutao) [inline clippy_lints/src/booleans.rs]: What about using sym! macro?

**Comment by Manishearth** [comment]
I'd rather not dynamically intern symbols. Perhaps we should use the sym macro for must_use and use a cast like in https://github.com/rust-lang/rust-clippy/pull/4673 for the other one?

**Comment by Manishearth** [comment]
We can also land https://github.com/rust-lang/rust-clippy/pull/4673 for now and improve it later

**Comment by phansch** [comment]
Let's land #4673 for now :+1:

**Pull request closed without merging**